src/ctl/ctl-sink.c

```diff
@@ -87,50 +89,46 @@ static void sink_handle_get_parameter(struct ctl_sink *s,
 		return cli_vERR(r);
 
 	/* wfd_content_protection */
-	if (rtsp_message_read(m, "{<>}", "wfd_content_protection") >= 0) {
-		r = rtsp_message_append(rep, "{&}",
-					"wfd_content_protection: none");
-		if (r < 0)
-			return cli_vERR(r);
-	}
+	check_and_response_option("wfd_content_protection", "none");
 	/* wfd_video_formats */
-	if (rtsp_message_read(m, "{<>}", "wfd_video_formats") >= 0) {
-		char wfd_video_formats[128];
-		sprintf(wfd_video_formats,
-			"wfd_video_formats: 00 00 03 10 %08x %08x %08x 00 0000 0000 10 none none",
+	char wfd_video_formats[128];
+	sprintf(wfd_video_formats, "wfd_video_formats: 00 00 03 10 %08x %08x %08x 00 0000 0000 10 none none",
 			s->resolutions_cea, s->resolutions_vesa, s->resolutions_hh);
-		r = rtsp_message_append(rep, "{&}", wfd_video_formats);
-		if (r < 0)
-			return cli_vERR(r);
-	}
+//	check_and_response_option("wfd_video_formats", wfd_video_formats);
+	check_and_response_option("wfd_video_formats", "40 00 01 10 0001bdeb 051557ff 00000fff 10 0000 001f 11 0780 0438, 02 10 0001bdeb 155557ff 00000fff 10 0000 001f 11 0780 0438");
 	/* wfd_audio_codecs */
-	if (rtsp_message_read(m, "{<>}", "wfd_audio_codecs") >= 0) {
-		r = rtsp_message_append(rep, "{&}",
-					"wfd_audio_codecs: AAC 00000007 00");
-		if (r < 0)
-			return cli_vERR(r);
-	}
+//	check_and_response_option("wfd_audio_codecs", "wfd_audio_codecs: AAC 00000007 00");
+	check_and_response_option("wfd_audio_codecs", "LPCM 00000003 00, AAC 0000000f 00, AC3 00000007 00");
 	/* wfd_client_rtp_ports */
-	if (rtsp_message_read(m, "{<>}", "wfd_client_rtp_ports") >= 0) {
-		char wfd_client_rtp_ports[128];
-		sprintf(wfd_client_rtp_ports,
-					"wfd_client_rtp_ports: RTP/AVP/UDP;unicast %d 0 mode=play", rstp_port);
-		r = rtsp_message_append(rep, "{&}",
-					wfd_client_rtp_ports);
-		if (r < 0)
-			return cli_vERR(r);
-	}
+	char wfd_client_rtp_ports[128];
+	sprintf(wfd_client_rtp_ports, "RTP/AVP/UDP;unicast %d 0 mode=play", rstp_port);
+	check_and_response_option("wfd_client_rtp_ports", wfd_client_rtp_ports);
+	check_and_response_option("wfd_content_protection", "none");
+	check_and_response_option("wfd_display_edid", "0001 00ffffffffffff0051f38f50010000000e100104a51d10ff2f0000a057499b2610484f000000010101010101010101010101010101011a36809c70381f403020350025a510000018000000fc00496e7465726e616c204c43440a000000fd003c3c9a9a0e00000000000000000000000000000000000000000000000000000030");
```

Comment:
This defines the sink screen. the machine used for reverse engineering has a 1920x1080 screen so that should be the info added here. Tweak to respect video formats from sink and restrictions from video parametes